substrate/frame/asset-conversion/src/migration.rs

```diff
@@ -0,0 +1,192 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Storage migrations.
+
+use super::*;
+use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+use log;
+#[cfg(feature = "try-runtime")]
+use sp_runtime::TryRuntimeError;
+use sp_std::marker::PhantomData;
+
+const LOG_TARGET: &'static str = "runtime::asset-conversion::migration";
+
+/// Module providing migration functionality for updating account ids when the type deriving a
+/// pool's account id from pool id has changed.
+pub mod new_pool_account_id {
+	use super::*;
+	use frame_support::traits::{
+		fungibles::{Refund as RefundT, ResetTeam as ResetTeamT},
+		tokens::Preservation,
+	};
+
+	/// Type facilitating the migration of existing pools to new account ids when the type deriving
+	/// a pool's account id from pool id has changed.
+	///
+	/// ### Parameters:
+	/// - `T`: The [`Config`] implementation for the target asset conversion instance with a new
+	///   account derivation method defined by [`PoolLocator`].
+	/// - `OldLocator`: The previously used type for account derivation.
+	/// - `ResetTeam`: A type used for resetting the team configuration of an LP token.
+	/// - `Refund`: A type used to perform a refund if the previous pool account holds a deposit.
+	/// - `WeightPerItem`: A getter returning the weight required for the migration of a single pool
+	///   account ID. It should include: 2 * weight_of(T::Assets::balance(..)) + 2 *
+	///   weight_of(T::Assets::transfer(..)) + 2 * weight_of(Refund::deposit(..)) + 2 *
+	///   weight_of(Refund::refund(..)) + weight_of(ResetTeam::reset_team(..));
+	pub struct Migrate<T, OldLocator, ResetTeam, Refund, WeightPerItem>(
+		PhantomData<(T, OldLocator, ResetTeam, Refund, WeightPerItem)>,
+	);
+	impl<T, OldLocator, ResetTeam, Refund, WeightPerItem> OnRuntimeUpgrade
+		for Migrate<T, OldLocator, ResetTeam, Refund, WeightPerItem>
+	where
+		T: Config<PoolId = (<T as Config>::AssetKind, <T as Config>::AssetKind)>,
+		OldLocator: PoolLocator<T::AccountId, T::AssetKind, T::PoolId>,
+		ResetTeam: ResetTeamT<T::AccountId, AssetId = T::PoolAssetId>,
+		Refund: RefundT<T::AccountId, AssetId = T::AssetKind>,
+		WeightPerItem: Get<Weight>,
```

Comment:
Or the type arguments above has to be requires in the benchmark config, and benchmark can be provided. I assume most of users wont need this migration.